rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/OneScanHelper.java

```diff
@@ -47,6 +47,7 @@ private void send(String action, Bundle extras, IntentResultCallback callback) {
             if (callback != null) {
                 extras.putBoolean(Constants.Result.ONESCAN_IS_NOT_INSTALLED, true);
                 callback.onResult(Activity.RESULT_CANCELED, extras);
+                mActivity.finish();
```

Comment:
@syedowaisali90 what does this fix, not sure I follow?

Reply:
@vincent-karuri It was working fine in a normal scenario i.e singe-scan. But when I enable batch-scan then it's not working as expected. Consider a case, when I scan the barcodes and pressed the back button then this line executes https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L193 and we cannot execute this line https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L198 Because the activity has finished before setting up the data. https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L92

Reply:
cool, then why do we need to call `mActivity.finish();` here? if it's called in this method https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L198

Reply:
`mActivity.finish();' this is for another scenario like if the one-scan app has not installed in the device then we set the result `Activity.RESULT_CANCELED` and finish up the activity.

Reply:
`setResultAndFinish(response);` never called in that scenario.

Reply:
actually, why do we have to place where we set result https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L194 and  https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L80 we should merge the two.

Reply:
if you notice when we enable a batch-scan then we continue the scanning and holding the data in the `dataArray` i.e `doScan` method called every time when `RESULT_OK` and whenever we pressed the back button then this method get called https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L191 and this time the result will be `RESULT_CANCELED` but we have a data in the `dataArray`

Reply:
but isn't the same method called for single scan?

Reply:
Yes this https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L191 method also called on single scan but this time result will `RESULT_OK`

Reply:
which is handled here https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L80

right? That's why I'm suggesting you merge this https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L194 with https://github.com/onaio/rdt-standard/blob/b372d7c3eceb507633ad90474a6afb37a394be05/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java#L86 since they both check for `RESULT_CANCELED`.

Reply:
fixed.